packages/shared/src/hooks/useWindowEvents.ts

```diff
@@ -1,14 +1,44 @@
-import { useEffect } from 'react';
+import { useCallback, useEffect } from 'react';
+import { AuthEvent } from '../lib/kratos';
 
-// to make intellisense work on the first parameter
-const useWindowEvents: typeof window.addEventListener = (
-  event: string,
-  func: (e: Event) => void,
+interface EventData {
+  eventKey?: AuthEvent | string;
+}
```

Comment:
This one actually also is implementation wide.

Reply:
I think this one has to be here since this is the base requirement of this specific hook.